.github/workflows/flake.yml

```diff
@@ -0,0 +1,39 @@
+name: Flake
+
+# Run flake on all PRs.  This ignores many pre-existing errors however should catch
+# regressions and allows us to remove ignores and results issues iteratively.
+
+on:
+  pull_request:
+    paths:
+      - '**/*.py'
+
+jobs:
+  flake8-lint:
+    runs-on: ubuntu-latest
+    name: Lint
+    steps:
+      - name: Check out source repository
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Set up Python environment
+        uses: actions/setup-python@v1
+        with:
+          python-version: "3.8"
+      - name: Add parser
+        run: echo -n "::add-matcher::ci/daos-flake-matcher.json"
```

Comment:
What does echoing this string (i.e. to stdout) do?

Reply:
This installs a "pattern matcher", which serves to translate pylint output into github annotations.

Reply:
Yes.  The major issue is that many of these tools work on the tree, and our existing linting tools are all setup to work on the patch only, and GHA has no easy way (or we've not found/written one) to limit comments to modified code.

Pylint for example has 2-3000 warnings for the daos src tree, in an ideal world we'd fix these of course, but in practice the horse has bolted so we're mainly trying to prevent regressions at this time.  Hence why this PR is so big, and so many checks are disabled.